osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/periodicmeterreads/GetPeriodicMeterReadsCommandExecutorTest.java

```diff
@@ -57,51 +54,108 @@
     private DlmsHelper dlmsHelper;
 
     @Mock
-    private AttributeAddressService attributeAddressService;
-
-    @Mock
-    private AmrProfileStatusCodeHelper amrProfileStatusCodeHelper;
+    private DlmsObjectConfigService dlmsObjectConfigService;
 
     @Mock
     private DlmsConnectionManager connectionManager;
 
+    private final DlmsDevice device = this.createDevice(Protocol.DSMR_4_2_2);
+    private final long from = 1111111L;
+    private final long to = 2222222L;
+    private final DateTime fromDateTime = new DateTime(this.from);
+    private final DateTime toDateTime = new DateTime(this.to);
+
     @Before
     public void setUp() {
         when(this.connectionManager.getDlmsMessageListener()).thenReturn(this.dlmsMessageListener);
     }
 
+    @Test
+    public void testExecuteNullRequest() throws ProtocolAdapterException {
+        try {
+            this.executor.execute(this.connectionManager, this.device, null);
+            fail("Calling execute with null query should fail");
+        } catch (final IllegalArgumentException e) {
+            // expected exception
```

Comment:
You can specify expected exceptions using `@Test(expected = IllegalArgumentException.class)`. You can also remove the `fail(..)` statement in that case.

For more fine grained controle you could use `org.junit.rules.ExpectedException`

Reply:
I know, I inherited this test and although it can be improved, it can be argued that this is part of this story.